### PR TITLE
Fixes chem dispenser button order

### PR DESCRIPTION
**Pull request by kriskog**
## About The Pull Request
Sorts the list of chems in the chem dispenser manually, removing the autosort that no longer works since chem id's are gone. Doing it manually once makes more sense than doing it automatically on every initialize anyway.
The observant among you might notice the sulphuric acid button has moved from its previous position, since it wasnt sorted properly before, I assume because "sacid".

Also speedmerge please, chemistry is suddenly very hard again. :grimacing:

## Why It's Good For The Game
Bug fix

## Changelog
:cl: Skoglol
fix: Chem dispenser buttons are now sorted alphabetically again.
/:cl:

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
Not sure if the sortList proc will play nicely with types but you should try using that on initialize and refreshparts and see if that does what you want without manual oversight.

**Comment by kriskog** [comment]
As for re-sorting on update_parts, please never allow this to happen. I work off of button placement memory, having buttons jumbled up every time someone emags or upgrades the dispenser would be horrible.

So the initialize used to use sortList already, which is what broke when the ids were replaced. The typepaths arent uniform anyway, so reagent/consumable etc would sort wrong if it sorted by the path alone. Closest we could get would be sortName, but it doesnt work out of the box. I assume we'd have to initialize the paths as vars to get that to work?


But really, is there a reason we should be sorting it instead of just doing it properly? I get that it makes it easier to add chems correctly to the dispenser, but honestly how often are we going to do that?

**Comment by ExcessiveUseOfCobblestone** [comment]
> But really, is there a reason we should be sorting it instead of just doing it properly? I get that it makes it easier to add chems correctly to the dispenser, but honestly how often are we going to do that?

manually sorting for a function output doesn't seem very proper, the only reason you should be sorting like this is for readability when reading the code itself. It doesn't have much to do with the frequency of how often are we going to edit this list.

Also the sortlist just needs a new cmp for reagents then.

```DM
/proc/cmp_reagents_asc(datum/reagent/a,datum/reagent/b)
    sorttext(initial(a.name),initial(b.name))
```

```DM
/obj/machinery/chem_dispenser/Initialize()
	. = ..()
	dispensable_reagents = sortList(dispensable_reagents, /proc/cmp_reagents_asc)
	update_icon()
```

**Comment by kriskog** [comment]
Allright, fair enough. The suggested cmp did it backwards so turned it around. Thanks

File: code/__HELPERS/cmp.dm [hunk 1]
```diff
@@ -102,3 +102,6 @@ GLOBAL_VAR_INIT(cmp_field, "name")
 
 /proc/cmp_job_display_asc(datum/job/A, datum/job/B)
 	return A.display_order - B.display_order
+
+/proc/cmp_reagents_asc(datum/reagent/a, datum/reagent/b)
+    return sorttext(initial(b.name),initial(a.name))
```
Comment (ExcessiveUseOfCobblestone) [inline code/__HELPERS/cmp.dm]: if b is first it's descending aka `_dsc`

**Review by ExcessiveUseOfCobblestone — DISMISSED** [review]

**Pull request merged**